### DISCUSSION TITLE
LaTeX full compiler and renderer support

### DISCUSSION THREAD

**Discussion by CrO2Cl2** [TriliumNext/Trilium#8263] [Plugin Ideas]
### Describe feature

I have been using the KaTeX math for a while, the option to fully integrate a LaTeX compiler and renderer as a Note type would be very useful, as I mainly use Overleaf for now, but would love to also be able to dysplay the LaTeX without using a PDF

**Comment by meichthys** [comment]
Unfortunately this will likely not happen as things currently stand. This feature would be dependent upon ckeditor5 which Trilium uses and the `MathType` editor plugin for ckeditor5 is a premium feature: https://ckeditor.com/docs/ckeditor5/latest/features/math-equations.html

**Comment by zerebos** [comment]
In theory it could be done as a separate note type, but as @meichthys said, it's very unlikely to be integrated to text notes due to it being locked behind premium.

**Comment by CrO2Cl2** [comment]
we already have the KaTeX for the text notes, I was thinking of just being able to import a .tex file and it gets rendered as it should, as this would make my school life 1000x easier

**Comment by zerebos** [comment]
Ah then that would definitely be a separate note, and this is something a plugin/widget of some kind could add. Doesn't sound too difficult in theory.

**Comment by CrO2Cl2** [comment]
and in practice? JS is one of the languages that I dont know anything about

**Comment by zerebos** [comment]
Well I cannot tinker around with it at the moment as I am at work, but I know it wouldn't be too difficult to do something similar to https://github.com/rauenzi/Trilium-MarkdownPreview but for KaTeX/LaTeX

**Comment by zerebos** [comment]
Definitely possible to do something like that

https://github.com/zadam/trilium/assets/6865942/0a365540-3a0b-45bf-ace3-a6af9a64a97a

**Comment by CrO2Cl2** [comment]
would you mind sharing the code that you Used for this?

**Comment by zerebos** [comment]
Sure, let me just clean up the code a bit first and I'll share it publicly

**Comment by zerebos** [comment]
@CrO2Cl2 Here, you can grab it (with instructions) here https://github.com/rauenzi/Trilium-LaTeXPreview#installation
